### PR TITLE
Fix: false negative of `indent` with `else if` statements (fixes #6956)

### LINKED ISSUE #6956: Inconsistent indentation allowed on chained if-else
**What version of ESLint are you using?**
v3.3.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
```js
{
  "env": {
    "node": true,
    "es6": true,
    "mocha": true
  },
  "rules": {
    "no-bitwise": 2,
    "curly": 0,
    "indent": [
      2,
      2,
      {
        "SwitchCase": 1
      }
    ],
    "no-caller": 2,
    "quotes": [
      2,
      "single"
    ],
    "no-undef": 2,
    "no-unused-vars": 2,
    "strict": 0,
    "max-len": [
      2,
      {
        "code": 100,
        "ignoreComments": true
      }
    ],
    "no-use-before-define": 0,
    "new-cap": 2,
    "no-sparse-arrays": 0,
    "no-trailing-spaces": 2
  },
  "parserOptions": {
    "sourceType": "module"
  }
}
```

**What did you do? Please include the actual source code causing the issue.**
```js
  if (type === '1') abacus.inc('rest.status.info');
    else if (type === '2') abacus.inc('rest.status.ok');
  else if (type === '4') abacus.inc('rest.status.clientError');
  else if (type === '5') abacus.inc('rest.status.serverError');
```
With the above if-else chain, I have the first else indented too far and we get the correct output: `error  Expected indentation of 2 space characters but found 4  indent`

But if the first else is correct, I find that all subsequent lines in the if-else chain don't generate errors when their indentation is off.

```js
  if (type === '1') abacus.inc('rest.status.info');
  else if (type === '2') abacus.inc('rest.status.ok');
               else if (type === '4') abacus.inc('rest.status.clientError');
      else if (type === '5') abacus.inc('rest.status.serverError');
```
This generates no output and linting passes.

**What did you expect to happen?**
I would expect the second input to generate indentation errors:
```sh
error  Expected indentation of 2 space characters but found 15  indent
error  Expected indentation of 2 space characters but found 6  indent
```

**What actually happened? Please include the actual, raw output from ESLint.**
Linting passes and generates no output.

### PR DESCRIPTION
**Pull request by not-an-aardvark**
**What issue does this pull request address?**

#6956

**What changes did you make? (Give an overview)**

Previously, only the first `else` statement in a chain would be checked for correct indentation. This PR fixes the issue by recursively checking nodes in `else if` chains to ensure that all nodes have the right indentation.

**Is there anything you'd like reviewers to focus on?**

Nothing in particular.

### CODE REVIEW

**Comment by gyandeeps** [comment]
Nice. 
LGTM, waiting for others to review.

File: tests/lib/rules/indent.js [hunk 1]
```diff
@@ -2435,6 +2435,45 @@ ruleTester.run("indent", rule, {
...
+            code:
+            "foo();\n" +
+            "  if (baz) foobar();\n" +
+            "  else qux();",
+            output:
+            "foo();\n" +
+            "if (baz) foobar();\n" +
+            "else qux();",
+            options: [2],
+            errors: expectedErrors([[2, 0, 2, "IfStatement"], [3, 0, 2, "Keyword"]])
```
Comment (vitorbal) [inline tests/lib/rules/indent.js]: Could you add some tests for when the if/else expression has braces as well?

**Comment by not-an-aardvark** [comment]
Added a few more tests.

**Comment by platinumazure** [comment]
LGTM. @vitorbal What do you think of the new tests?

**Comment by vitorbal** [comment]
Thanks, @not-an-aardvark! LGTM as well.

**Comment by ilyavolodin** [comment]
LGTM. Thanks for contributing, @not-an-aardvark

**Pull request merged**